src/components/Checkbox/index.tsx

```diff
@@ -1,43 +1,41 @@
 import React, { useCallback } from 'react';
-import {
-    MdCheckBox,
-    MdCheckBoxOutlineBlank,
-    MdIndeterminateCheckBox,
-} from 'react-icons/md';
 import { _cs } from '@togglecorp/fujs';
+
+import DefaultCheckmark, { CheckmarkProps } from '../Checkmark';
 import { UiMode } from '../ThemeContext';
 import VisualFeedback from '../VisualFeedback';
 import { useThemeClassName } from '../../hooks';
 
 import styles from './styles.css';
 
-export interface CheckboxProps<T, K> {
+export interface CheckboxProps<K, N> {
     className?: string;
     labelClassName?: string;
-    checkIconClassName?: string;
-    label?: string | number;
+    checkmark?: (p: CheckmarkProps) => React.ReactElement;
+    checkmarkClassName?: string;
+    label?: React.ReactNode;
     disabled?: boolean;
     readOnly?: boolean;
-    // NOTE: if value is false and indeterminate is true, show a filled checkbox
     indeterminate?: boolean;
     uiMode?: UiMode;
     tooltip?: string;
-    value: T;
-    onChange: (value: T, name: K) => void;
-    name: K;
+    value: K;
+    onChange: (value: K, name: N) => void;
+    name: N;
 }
 
-function Checkbox<T extends boolean, K extends string>(props: CheckboxProps<T, K>) {
+function Checkbox<K extends boolean, N extends string>(props: CheckboxProps<K, N>) {
```

Comment:
Why do we even need the type `K`? Why not just use `boolean`?

Reply:
More strict typing, just to assure that same value is returned via onChange